keras_cv/models/backbones/efficientnet_v1/efficientnet_v1_backbone.py

```diff
@@ -0,0 +1,484 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""EfficientNet V1 models for Keras.
+
+Reference:
+    - [EfficientNet: Rethinking Model Scaling for Convolutional Neural Networks](https://arxiv.org/abs/1905.11946)
+        (ICML 2019)
+    - [Based on the original keras.applications EfficientNet](https://github.com/keras-team/keras/blob/master/keras/applications/efficientnet.py)
+"""  # noqa: E501
+
+import copy
+import math
+
+from tensorflow import keras
+from tensorflow.keras import backend
+from tensorflow.keras import layers
+
+from keras_cv.models import utils
+from keras_cv.models.backbones.backbone import Backbone
+from keras_cv.models.backbones.efficientnet_v1.efficientnet_v1_backbone_presets import (  # noqa: E501
+    backbone_presets,
+)
+from keras_cv.utils.python_utils import classproperty
+
+
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class EfficientNetV1Backbone(Backbone):
+    """This class represents a Keras EfficientNet architecture.
+    Args:
+        include_rescaling: bool, whether to rescale the inputs. If set to
+            True, inputs will be passed through a `Rescaling(1/255.0)` layer.
+        width_coefficient: float, scaling coefficient for network width.
+        depth_coefficient: float, scaling coefficient for network depth.
+        dropout_rate: float, dropout rate before final classifier layer.
+        drop_connect_rate: float, dropout rate at skip connections.
+        depth_divisor: integer, a unit of network width.
```

Comment:
This arg needs better documentation

Reply:
Should I write something like? 

width_coefficient: float, controls the scaling coefficient for the network width. Increasing this value will result in a wider network, while decreasing it will result in a narrower network.

depth_coefficient: float, controls the scaling coefficient for the network depth. Increasing this value will result in a deeper network, while decreasing it will result in a shallower network.

dropout_rate: float, specifies the dropout rate before the final classifier layer. It helps prevent overfitting.

drop_connect_rate: float, specifies the dropout rate at skip connections. Setting a dropout rate for skip connections helps regularize the network and prevent overfitting.

depth_divisor: integer, determines a unit of network width. It is used to ensure that the number of filters in each layer is divisible by this value and is used for achieving better computational efficiency.

Reply:
Hey, pretty good! I was actually only talking about `depth_divisor`. Your clarification is helpful, but I still don't know if a user reading this would know good values to use.

Reply:
New idea: since this param is `8` for all presets maybe that should be the default value?

Reply:
Yeah, I think this will be good. Also, the default value for drop_connect_rate is 0.2. Adding this as well.